tests/mock_crust/churn.rs

```diff
@@ -328,115 +356,115 @@ fn churn() {
+fn messages_during_churn() {
     let min_section_size = 8;
     let network = Network::new(min_section_size, None);
     let mut rng = network.new_rng();
-
-    let mut nodes = if initial_nodes == 1 {
-        Nodes(vec![TestNode::builder(&network).first().endpoint(Endpoint(0)).create()])
-    } else {
-        create_connected_nodes(&network, initial_nodes)
+    let mut nodes = create_connected_nodes(&network, 20);
+    let mut clients = create_connected_clients(&network, &mut nodes, 1);
+    let cl_auth = Authority::Client {
+        client_key: *clients[0].full_id.public_id().signing_public_key(),
+        proxy_node_name: nodes[0].name(),
+        peer_id: clients[0].handle.0.borrow().peer_id,
     };
 
-    while nodes.len() < min_section_size {
-        let (added_index, _) = add_random_node(&mut rng, &network, &mut nodes, min_section_size);
-        let _ = poll_all(&mut nodes, &mut []);
+    for i in 0..100 {
+        trace!("Iteration {}", i);
+        let added_index = random_churn(&mut rng, &network, &mut nodes);
+
+        // Create random data ID and pick random sending and receiving nodes.
+        let data_id = DataIdentifier::Immutable(rng.gen());
+        let index0 = gen_range_except(&mut rng, 0, nodes.len(), added_index);
+        let index1 = gen_range_except(&mut rng, 0, nodes.len(), added_index);
+        let auth_n0 = Authority::ManagedNode(nodes[index0].name());
+        let auth_n1 = Authority::ManagedNode(nodes[index1].name());
+        let auth_g0 = Authority::NaeManager(rng.gen());
+        let auth_g1 = Authority::NaeManager(rng.gen());
+        let section_name: XorName = rng.gen();
+        let auth_s0 = Authority::Section(section_name);
+        // this makes sure we have two different sections if there exists more than one
+        // let auth_s1 = Authority::Section(!section_name);
+
+        let mut expected_gets = ExpectedGets::default();
+
+        // Test messages from a node to itself, another node, a group and a section...
+        expected_gets.send_and_expect(data_id, auth_n0, auth_n0, &mut nodes, min_section_size);
+        expected_gets.send_and_expect(data_id, auth_n0, auth_n1, &mut nodes, min_section_size);
+        expected_gets.send_and_expect(data_id, auth_n0, auth_g0, &mut nodes, min_section_size);
+        expected_gets.send_and_expect(data_id, auth_n0, auth_s0, &mut nodes, min_section_size);
+        // ... and from a group to itself, another group, a section and a node...
+        expected_gets.send_and_expect(data_id, auth_g0, auth_g0, &mut nodes, min_section_size);
+        expected_gets.send_and_expect(data_id, auth_g0, auth_g1, &mut nodes, min_section_size);
+        expected_gets.send_and_expect(data_id, auth_g0, auth_s0, &mut nodes, min_section_size);
+        expected_gets.send_and_expect(data_id, auth_g0, auth_n0, &mut nodes, min_section_size);
+        // ... and from a section to itself, another section, a group and a node...
+        // TODO: Enable these once MAID-1920 is fixed.
```

Comment:
yeh but this is not the churn test in master, this is the test which was in master before it got updated and at that time it was still disabled. Going to leave this ignored for now. If fine to uncomment we can take that in a new PR.